### PR TITLE
[Import/Export] Exported databases can now be safely moved

### LINKED ISSUE #5868: Absolute paths in EXPORT DATABASE load.sql
### What happens?

If you `EXPORT DATABASE 'mypath' (FORMAT PARQUET);` the **absolute paths** to `mypath` are written in to `load.sql`.

This means you can't copy the exported files to another location and  `IMPORT DATABASE 'mypath'` because you'll get:
```
duckdb.InvalidInputException: Invalid Input Error: Attempting to execute an unsuccessful or closed pending query result
Error: IO Error: No files found that match the pattern "/original/path/table.parquet"
```

### To Reproduce

1. `EXPORT DATABASE 'test' (FORMAT PARQUET);`

1. Open `test/load.sql`

1. Observe full paths in `test` directory, e.g.: `/home/dave/backup/test/table.parquet`

1. Copy `test` to another location e.g. `test2`

1. `IMPORT DATABASE 'test2'`

1. Observe `IO Error`

### OS:

Ubuntu 20.04.5 LTS

### DuckDB Version:

v0.6.1-dev83 dfae126

### DuckDB Client:

Python

### Full Name:

Dave Tapley

### Affiliation:

JE Fuller

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #5868 

Previously we would EXPORT and bake the absolute path of the table data into the query e.g `path/to/exported/db/tbl.csv`
When we only really need the `tbl.csv` part, because the path of the `db` folder is already provided on IMPORT.

Now we prefix the path with a placeholder on EXPORT, and replace it with the `db` folder path on IMPORT.

### CODE REVIEW

**Comment by Tishj** [comment]
Actually, do we even need the placeholder?

**Comment by Mytherin** [comment]
Thanks for the PR!

While the placeholder approach is relatively straightforward I do see some potential issues that with this:

* It won't work correctly with older versions of DuckDB - as they won't know how to handle a placeholder, and won't emit a placeholder
* It won't work when loading into other systems, e.g. Postgres
* Global text replacement is a bit hacky, since the `IMPORT` file can contain arbitrary user-defined input (e.g. the actual file name)

Perhaps we can use the Parser for this - for example we can run the parser on the `load.sql` file. It should only contain `COPY` statements - and we can directly extract and modify the path for each of the `CopyStatements` within the file. We could then implement `CopyStatement::ToString` (similar to what we have for `SelectStatement`) to stringify it again. That should work while leaving the export code untouched.

**Comment by Tishj** [comment]
You mean from new version -> old version
Is that supported in general?

Because for old version -> new version this should still work
it doesn't assert that the file is prefixed with the placeholder, only when it does it replaces it

The other points are fair though
I like the approach you suggested, I felt the string replacements were kinda dirty/error-prone to begin with

**Comment by Mytherin** [comment]
> You mean from new version -> old version
> Is that supported in general?

It used to work, so I would prefer not to break it unnecessarily

**Comment by Tishj** [comment]
I've made some changes to the statement verifier to include the copy statement

Though I went a little overboard and implemented Equals for every Statement
I think I'll make a separate PR out of it, and also add Copy + Equals verification for all statements

Found and fixed a few bugs in this ToString that I need to move over to this PR, but other than that it should be done

File: src/parser/statement/copy_statement.cpp [hunk 1]
```diff
@@ -11,6 +11,101 @@ CopyStatement::CopyStatement(const CopyStatement &other) : SQLStatement(other),
...
+			}
+			result += KeywordHelper::WriteOptionallyQuoted(info.select_list[i]);
+		}
+		result += " )";
+	}
+	return result;
+}
+
+string CopyStatement::ToString() const {
+	string result;
```
Comment (Mytherin) [inline src/parser/statement/copy_statement.cpp:86]: We have verification for `ToString` round-trips for INSERT/DELETE/SELECT statements in `src/main/client_context.cpp:670` (in `PendingStatementOrPreparedStatementInternal`). Perhaps add the COPY statement there as well?

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**